### PR TITLE
Rhumblines

### PR DESCRIPTION
**Pull request by S-Man42**

### CODE REVIEW

**Comment by Mike-3** [comment]
solve conflics

**Comment by S-Man42** [comment]
Done.

**Comment by S-Man42** [comment]
#1549

File: lib/tools/coords/map_view/logic/map_geometries.dart [hunk 1]
```diff
@@ -55,25 +56,42 @@ class GCWMapLine extends GCWMapSimpleGeometry {
...
+        _calculateGeodetic(); break;
+      case GCWMapLineType.RHUMB:
+        _calculateRhumb(); break;
+      default:
+        _calculateGeodetic(); break;
+    }
+  }
+
+  void _calculateRhumb() {
+    DistanceBearingData _distBear = rhumbline.distanceBearing(start.point, end!.point, defaultEllipsoid);
```
Comment (Mike-3) [inline lib/tools/coords/map_view/logic/map_geometries.dart:85]: null safety ?

File: lib/tools/coords/map_view/logic/map_geometries.dart [hunk 2]
```diff
@@ -55,25 +56,42 @@ class GCWMapLine extends GCWMapSimpleGeometry {
...
+    length = _distBear.distance;
+    bearingAB = _distBear.bearingAToB;
+    bearingBA = _distBear.bearingBToA;
+
+    shape.add(end!.point);
   }
 
   void _calculateGeodetic() {
-    DistanceBearingData _distBear = distanceBearing(start.point, end!.point, defaultEllipsoid);
+    DistanceBearingData _distBear = geodetic.distanceBearing(start.point, end!.point, defaultEllipsoid);
```
Comment (Mike-3) [inline lib/tools/coords/map_view/logic/map_geometries.dart:94]: null safety ?

File: lib/tools/coords/map_view/persistence/model.dart [hunk 3]
```diff
@@ -104,19 +107,25 @@ class MapPointDAO {
...
+  MapPolylineDAO(this.uuid, this.pointUUIDs, this.color, this.type);
 
-  Map<String, Object?> toMap() => {'uuid': uuid, 'pointUUIDs': pointUUIDs, 'color': color};
+  Map<String, Object?> toMap() => {'uuid': uuid, 'pointUUIDs': pointUUIDs, 'color': color, 'type': type};
 
   MapPolylineDAO.fromJson(Map<String, Object?> json)
       : uuid = toStringOrNull(json['uuid']) ?? const Uuid().v4(),
         pointUUIDs = List<String>.from(toStringListOrNull(json['pointUUIDs']) ?? []),
-        color = toStringOrNull(json['color']) ?? colorToHexString(COLOR_MAP_POLYLINE);
+        color = toStringOrNull(json['color']) ?? colorToHexString(COLOR_MAP_POLYLINE),
```
Comment (Mike-3) [inline lib/tools/coords/map_view/persistence/model.dart:127]: is is not better, make the strings type, color, uuid, .. to constants ?

File: lib/tools/coords/_common/logic/external_libs/net.sf.geographic_lib/geographic_lib/geodesic.dart [hunk 4]
```diff
@@ -228,12 +228,12 @@ class Geodesic {
...
   //   tiny_ + epsilon() == epsilon()
-  static final double tiny_ = sqrt(double.minPositive);
-  static const double _tol0_ = double.minPositive;
+  static final double tiny_ = sqrt(positiveDoublePrecision);
+  static double _tol0_ = positiveDoublePrecision;
   // Increase multiplier in defn of tol1_ from 100 to 200 to fix inverse case
   // 52.784459512564 0 -52.784459512563990912 179.634407464943777557
   // which otherwise failed for Visual Studio 10 (Release and Debug)
-  static const double _tol1_ = 200 * _tol0_;
+  static double _tol1_ = 200 * _tol0_;
```
Comment (Mike-3) [inline lib/tools/coords/_common/logic/external_libs/net.sf.geographic_lib/geographic_lib/geodesic.dart:236]: to final or const

File: lib/tools/coords/_common/logic/external_libs/net.sf.geographic_lib/geographic_lib/aux_latitude.dart [hunk 5]
```diff
@@ -0,0 +1,1420 @@
...
+    }
+    return _AuxAngleNiter(_AuxAngle(tphi).copyquadrant(zeta), niter);
+  }
+
+  void fillcoeff(int auxin, int auxout, int k) {
+    List<double> coeffs = [];
+    List<int> ptrs = [];
+  
+    if (_GEOGRAPHICLIB_AUXLATITUDE_ORDER == 4) {
+      coeffs = [
```
Comment (Mike-3) [inline lib/tools/coords/_common/logic/external_libs/net.sf.geographic_lib/geographic_lib/aux_latitude.dart:533]: coeffs and ptrs lists mayby to const ?

**Review by Mike-3 — CHANGES_REQUESTED** [review]

**Comment by Mike-3** [comment]
Because of dart_numerics, a web version can no longer be created.

**Comment by S-Man42** [comment]
f**k...

**Comment by S-Man42** [comment]
ok, will change.

**Pull request merged**